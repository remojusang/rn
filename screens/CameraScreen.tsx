import React from 'react';
import type { NativeStackScreenProps } from '@react-navigation/native-stack';
import { RootStackParamList } from '../utils/types';
import ScreenLayout from '../components/ScreenLayout';
import CameraModules from '../components/CameraModules';

type CameraProps = NativeStackScreenProps<
  RootStackParamList,
  'Camera'
>;

function CameraScreen({}: CameraProps) {
  return (
    <ScreenLayout isLoading={false}>
      <CameraModules />
    </ScreenLayout>
  );
}

export default CameraScreen;
