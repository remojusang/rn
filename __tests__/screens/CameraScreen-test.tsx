import { render } from '@testing-library/react-native';
import CameraScreen from '../../screens/CameraScreen';
import ErrorBoundary from '../../components/MyErrorBoundary';
// 에러 방지 => react-native-permissions: NativeModule.RNPermissions is null
jest.mock('react-native-permissions', () =>
  require('react-native-permissions/mock'),
);

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('CameraScreen', () => {
  test('렌더링 테스트', () => {
    const TEST_PROPS: any = {
      route: jest.fn(),
      navigation: jest.fn(),
    };
    const renderedJson = render(
      <ErrorBoundary>
        <CameraScreen {...TEST_PROPS} />,
      </ErrorBoundary>,
    ).toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
});
