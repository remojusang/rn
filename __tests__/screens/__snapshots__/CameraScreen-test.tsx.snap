// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraScreen 렌더링 테스트 1`] = `
<div
  className="error-screen"
>
  <h2>
    An error has occured
  </h2>
  <h4>
    [React Intl] Could not find required \`intl\` object. &lt;IntlProvider&gt; needs to exist in the component ancestry.
  </h4>
</div>
`;
